docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,250 @@ sidebar_position: 7
+- fix bugs you might have missed (it always helps to get a second pair of eyes!)
+- edit for consistency with the rest of the codebase
+- discuss an unforseen issue with the code you're adding
 
-### 5. Pulling from upstream
+...
 
-When you go back to the codebase, it's likely that more changes will have been made on the original BobaBoard codebase your code is forked from. To update your git fork to reflect the changes of "upstream" (aka the original codebase) run the following command.
+There will be a bit of back and forth, and you might get comments requesting to change some of your code or asking clarifying questions. This is normal and expected: no engineer, no matter how senior, never gets comments and questions on their code. It's just part of the job, and not a reflection of your skills.
 
-```bash   showLineNumbers
-git pull upstream master
-```
+Most often, code reviews are just a matter of the maintainer enforcing consistency in the codebase, or spotting bugs you might have missed (like we all do), or asking to make code clearer for someone that has never read it before. Always feel free to ask clarifying questions, or to push back if you disagree with something.
 
-You can now go back to step 1 and make new changes all over again.
+Once your pull request is merged (aka accepted in the main codebase), you can delete your branch. Github has a easy button for you to do so.
 
-## Extra: Use a different GitHub account
+#### How to Make Requested Changes
 
-If you already have a GitHub account set up on your machine and want to use a different one for BobaBoard, you can use the following instructions:
+When changes are requested, you will likely see comments on the pull request you have submitted. Simply go back to step 2, write code, commit and push until all comments are fixed. Then go back to the pull request, check that everything is in order, and ask for an additional review. You should mark the comments you have addressed as done by using the "resolve convo" button.
 
-### 0. Create a special folder to house BobaBoard code
+### 7. Pulling from upstream
 
-This is the folder you're going to `clone` the BobaBoard repositories in.
+Now that the feature code has been merged to the `upstream master` branch, the code in the `master` branch of the developer's fork (which doesn't contain the original feature) is out of date. To download the updated code, the developer "pulls" the changes from the `upstream` codebase. This copies all the new code commits in the `upstream` codebase to the developer's.
 
-### 1. Set Up your Git username and email
+...
 
-In your new folder, run the following commands to set up the new user:
+When you go back to the codebase, it's likely that more changes will have been made on the original BobaBoard codebase your code is forked from. To update your git fork to reflect the changes of "upstream" (aka the original codebase) run the following command.
```

Comment:
I'm a little confused by the wording here. I think you're talking about other people having made changes, but it's not super clear. If that's what you're saying I'd do something like:

As you work, it's very likely other people will have made pull requests that will have been accepted to the original codebase you've forked from. To update your own fork and local copies to reflect the changes of "upstream" (the original codebase) run: